tools/migrate/migrations/20241119154436_plan.up.sql

```diff
@@ -0,0 +1,5 @@
+-- drop index "planphase_plan_id_key" from table: "plan_phases"
+DROP INDEX "planphase_plan_id_key";
+-- create index "planphase_plan_id_key_deleted_at" to table: "plan_phases"
+-- atlas:nolint
+CREATE UNIQUE INDEX "planphase_plan_id_key_deleted_at" ON "plan_phases" ("plan_id", "key", "deleted_at");
```

Comment:
We discussed offline that this allows multiple plans with same id & key as long as they all have deletedAt NULL

Reply:
Yup. We need to upgrade to Postgres 15+ which supports `NULLS NOT DISTINCT` which is the proper fix for this issue.